config/packages/messenger.yaml

```diff
@@ -12,13 +11,19 @@ framework:
                 retry_strategy:
                     max_retries: 3
                     multiplier: 2
+            email:
+                dsn: '%env(MESSENGER_TRANSPORT_DSN)%'
+                options:
+                    queue_name: email
+            sms:
+                dsn: '%env(MESSENGER_TRANSPORT_DSN)%'
+                options:
+                    queue_name: sms
             failed: 'doctrine://default?queue_name=failed'
             # sync: 'sync://'
 
         routing:
-            Symfony\Component\Mailer\Messenger\SendEmailMessage: async
+            Symfony\Component\Mailer\Messenger\SendEmailMessage: email
```

Comment:
You have the infinite loop I was talking about, don't you? Did you receive an email with this implementation?